src/pages/ResearcherProfile.js

```diff
@@ -272,74 +105,60 @@ export default function ResearcherProfile(props) {
     const currentUserUpdate = Storage.getCurrentUser();
     delete currentUserUpdate.email;
     currentUserUpdate.displayName = profile.profileName;
-    currentUserUpdate.additionalEmail = profile.additionalEmail;
     currentUserUpdate.roles = userRoles;
-    currentUserUpdate.institutionId = profile.institutionId;
-    const payload = { updatedUser: currentUserUpdate };
-    let updatedUser = await User.update(payload, currentUserUpdate.userId);
-    updatedUser = Object.assign({}, updatedUser, setUserRoleStatuses(updatedUser, Storage));
-    return updatedUser;
-  };
-
-  const cloneProfile = (profile) => {
-    return omit(cloneDeep(profile), ['libraryCards', 'libraryCardEntries']);
-  };
 
-  const generateCountryNames = () => {
-    const USA = option({ value: 'United States of America'}, ['United States of America']);
-    const empty = option({ value: ''}, ['']);
-    const countryNames = getNames().map((name) => option({value: name}, [name]));
-    const index = countryNames.indexOf(USA);
-    countryNames.splice(index, 1);
-    countryNames.splice(0, 0, USA);
-    countryNames.splice(0, 0, empty);
-    setCountryNames(countryNames);
-  };
+    currentUserUpdate.eraCommonsId = profile.eraCommonsId;
+    currentUserUpdate.institutionId = profile.institutionId;
+    currentUserUpdate.suggestedInstitution = profile.suggestedInstitution;
+    currentUserUpdate.selectedSigningOfficialId = profile.selectedSigningOfficialId;
+    currentUserUpdate.suggestedSigningOfficial = profile.suggestedSigningOfficial;
 
-  const generateStateNames = () => {
-    const empty = option({ value: ''}, ['']);
-    const UsaStates = require('usa-states').UsaStates;
-    const stateNames = (new UsaStates().arrayOf('names')).map((name) => option({value: name}, [name]));
-    stateNames.splice(0, 0, empty);
-    setStateNames(stateNames);
-  };
+    const payload = currentUserUpdate;
 
-  const showIncompleteFields = () => {
-    const listIncompleteFields = incompleteFields.map((field) =>
-      li({ key: field }, [field])
-    );
-    return ul({}, [listIncompleteFields]);
+    let updatedUser = await User.updateSelf(payload);
```

Comment:
If you go up to line 105, the payload here is derived from `Storage.getCurrentUser()` so it has lots of extra information that we don't really want to be sending as a payload in this API call. I would suggest that we build a generic payload object and populate it with the fields we care about.

Reply:
great point! now sends a custom payload object.